src/platform-aws.c

```diff
@@ -37,9 +38,15 @@ struct ec2_platform_data {
 	int domain_per_thread;
 };
 
+/*
+ * platform_data_map is an ordered list of platform entries.  The
+ * name is evaluated as a POSIX regex, so entries like "p5.*" is
+ * valid.  First match found wins, even if there is a more specific
+ * match later in the array.
+ */
 static struct ec2_platform_data platform_data_map[] = {
 	{
-		.name = "p4d.24xlarge",
+		.name = "^p4d.24xlarge$",
```

Comment:
totally optional bike-shedding suggestion: convert `name` to `name_re`/`name_regex` to make it even more self-documenting